### PR TITLE
Update example to template hash and version

### PR DESCRIPTION
**Pull request by mwcraig**

### CODE REVIEW

**Comment by jakirkham** [comment]
Did you mean to fork? 😉

**Comment by mwcraig** [comment]
yeah, I just clicked the pencil, sorry about that....

**Comment by mwcraig** [comment]
@conda-forge/core -- can you take a look and let me know what other changes are needed?

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -9,7 +11,7 @@ source:
     fn: simplejson-{{ version }}.tar.gz
     url: https://pypi.io/packages/source/s/simplejson/simplejson-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/example/meta.yaml]: Do we want to start templating the name too? I've found this handy in a few cases and other people seem to have picked up on it. Though this suggestion may be out of scope for this proposal.

Comment (jakirkham) [inline recipes/example/meta.yaml]: See this [example]( https://github.com/conda-forge/funcsigs-feedstock/blob/f896b21228dd6dcaa74c2c60cad9b3f1b4648ebb/recipe/meta.yaml ) to get an idea of what I mean.

File: recipes/example/meta.yaml [hunk 2]
```diff
@@ -9,7 +11,7 @@ source:
...
     # md5 is OK but sha256 is desirable.
-    sha256: d58439c548433adcda98e695be53e526ba940a4b9c44fb9a05d92cd495cdd47f
+    {{ hash_type }}: {{ hash_val }}
```
Comment (pelson) [inline recipes/example/meta.yaml]: Personally I wouldn't bother encapsulating the `hash_type`.

Comment (jakirkham) [inline recipes/example/meta.yaml]: Yeah, it seems static enough. Though I haven't argued with anyone that went that route.

Comment (pelson) [inline recipes/example/meta.yaml]: No indeed - I agree with not having a strictly enforced policy on this. Guidance on what makes a maintainable recipe is all we should strive for.

File: recipes/example/meta.yaml (hunk 1 above)
Comment (mwcraig) [inline recipes/example/meta.yaml]: My leaning is keep the sample a little simpler, but I'm hoping to add another example today too.

**Pull request merged**